genie/load.py

```diff
@@ -239,10 +241,10 @@ def _generate_primary_key(
     Returns:
         pd.DataFrame: The dataframe with primary_key column added
     """
-    # replace NAs with emtpy string
-    dataset = dataset.fillna("")
     # generate primary key column for original database
     dataset[primary_key_cols] = dataset[primary_key_cols].applymap(str)
+    # replace NAs with emtpy string
+    dataset[primary_key_cols] = dataset[primary_key_cols].fillna("")
```

Comment:
So if `applymap` converts int to float and then applies `str`, what is the goal of this later line if we don't want to add back the `.0`?